app/main/views/frameworks.py

```diff
@@ -347,6 +348,7 @@ def framework_submission_services(framework_slug, lot_slug):
 @main.route('/frameworks/<framework_slug>/declaration/start', methods=['GET'])
 @login_required
 @EnsureApplicationCompanyDetailsHaveBeenConfirmed(data_api_client)
```

Comment:
hmmmmmmmm interesting how the decorator above *doesn't* use the lambda trick. wonder how it handles mocking in tests...

Reply:
It mocks out it's `validator` method, which bypasses having to use the apiclient at all.

Reply:
Weird. We're really going to have to sort out these apiclients and put a single one on each request context at some point. Or rather the `g` object.